pulp_container/app/tasks/builder.py

```diff
@@ -0,0 +1,132 @@
+from pulp_container.app.models import (
+    Blob,
+    BlobManifest,
+    ContainerRepository,
+    Manifest,
+    Tag
+)
+from pulp_container.constants import MEDIA_TYPE
+from pulpcore.app.models.content import Artifact, ContentArtifact
+from pulpcore.plugin.tasking import WorkingDirectory
+
+import json
+import os
+import shutil
+import subprocess
+from uuid import uuid4
+
+
+def get_or_create_blob(layer_json, manifest, path):
+    """
+    Creates Blob from json snippet of manifest.json
+
+    Args:
+        layer_json (json): json
+        path (str): Path of the directory that contains layer
+
+    Returns:
+        class:`pulp_container.app.models.Blob`
+
+    """
+    try:
+        return Blob.objects.get(digest=layer_json['digest'])
```

Comment:
i think we need to make a relation here between the manifest and the found blob.

Reply:
You are right. I'll refactor the code to always create that relation.